snyk/snyk
# feat: yaml support for iac wasm bundles [CFG-1271]

#### What does this PR do?

This PR bumps @open-policy-agent/opa-wasm to [1.6.0](https://github.com/open-policy-agent/npm-opa-wasm/releases/tag/1.6.0) to include support for the `yaml.unmarshal()` function in Rego and support for Node 10. This allows the `snyk iac` command to support additional policies that unmarshall YAML content within Terraform configuration.

#### Where should the reviewer start?

package.json is the primary change here but the implementation of `yaml.unmarshal()` is here: https://github.com/open-policy-agent/npm-opa-wasm/pull/100

This also removes the use of the `mock-fs` package in the file-scanner.spec.ts test as the opa-wasm package will use `console.error` to write out a warning to stderr when loading the wasm fixture and this somehow causes `mock-fs` to throw the following error bringing the test suite down:

```
ENOENT: no such file or directory, lstat '/Users/Aron/Code/snyk/node_modules/@jest/console/node_modules'
``` 

Rather than debug this weird behavior we've decided to just mock out the function that loads the fixtures directly in the test suite.

#### How should this be manually tested?

[rules.zip](https://github.com/snyk/snyk/files/7719060/rules.zip)

Download the above zip file containing a custom rules bundle and unzip:

With the `snyk-iac-rules` tool build a custom bundle:
```
% snyk-iac-rules build
```

Then test the fixture with the latest snyk cli:

```
% snyk-dev iac test --rules=bundle.tar.gz rules/HELLO/fixtures/sg.tf
```

You should see 15 issues found vs. 14 when run with current snyk.

#### What are the relevant tickets?

[CFG-1271]



[CFG-1271]: https://snyksec.atlassian.net/browse/CFG-1271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
